### PR TITLE
new custom vpn section

### PR DESCRIPTION
**Pull request by cotosso**
This section explain how to load and use a custom vpn file, usually these configuration files come by a third-party VPN provider.

### CODE REVIEW

File: custom_openvpn_tunnel.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+The configuration ensures the VPN starts automatically when the firewall boots.
+
+Prerequisites
+-------------
+
+- A valid OpenVPN configuration file (`myvpn.ovpn`) from your VPN provider.
+- Access to the NethSecurity terminal via SSH.
+- Basic familiarity with the UCI (Unified Configuration Interface) system in OpenWrt/NethSecurity.
+
+Additional Notes on CLI Configuration
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:17]:
```suggestion
Additional notes on CLI configuration
```

File: custom_openvpn_tunnel.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+-------------------------------------
+
+- This procedure does not include any validation of the entered data. Therefore, it is intended to be performed by advanced users familiar with the NethSecurity environment and OpenVPN configurations.
+- The VPN created using this method will not appear in the NethSecurity web interface and can only be managed through the command line interface (CLI).
+- It is critical to avoid using the same name for a VPN created via the CLI and one configured through the NethSecurity web interface. Since there are no safeguards in place to prevent naming conflicts, such overlap may cause configuration issues.
+
+For these reasons, caution and attention to detail are strongly advised when performing this procedure.
+
+Steps to Configure the VPN
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:26]:
```suggestion
Steps to configure the VPN
```

File: custom_openvpn_tunnel.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+--------------------------
+
+1. Place the Configuration File in the Correct Directory
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:29]:
```suggestion
1. Place the configuration file in the correct Directory
```

File: custom_openvpn_tunnel.rst [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+2. Verify the permissions for the file, the output should look like this: ::
+
+    ls -l /etc/openvpn/myvpn.ovpn
+    -rw-r--r--    1 root     root        <size> myvpn.ovpn
+
+If it is not so then set the correct permissions: ::
+
+    chmod 644 /etc/openvpn/myvpn.ovpn
+    chown root:root /etc/openvpn/myvpn.ovpn
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:45]:
```suggestion
2. Make sure to set the correct file permissions: ::

    chmod 644 /etc/openvpn/myvpn.ovpn
    chown root:root /etc/openvpn/myvpn.ovpn
```

File: custom_openvpn_tunnel.rst [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+3. Create a New OpenVPN Client Configuration in UCI
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:47]:
```suggestion
3. Create a new OpenVPN Client configuration in UCI
```

File: custom_openvpn_tunnel.rst [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    uci add openvpn openvpn
+    uci rename openvpn.@openvpn[-1]='myvpn'
+    uci set openvpn.myvpn.enabled='1'
+    uci set openvpn.myvpn.config='/etc/openvpn/myvpn.ovpn'
+
+2. Commit the changes to save the configuration: ::
+
+    uci commit openvpn
+   
+4. Start the VPN Client Immediately
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:61]:
```suggestion
4. Start the VPN client immediately
```

File: custom_openvpn_tunnel.rst [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To start the VPN client without rebooting the system, run: ::
+
+    /etc/init.d/openvpn restart
+
+This will restart all configured openvpn tunnels.
+
+5. Verify the VPN is Running
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:69]:
```suggestion
5. Verify the VPN is running
```

File: custom_openvpn_tunnel.rst [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+    uci commit openvpn
+   
+4. Start the VPN Client Immediately
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To start the VPN client without rebooting the system, run: ::
+
+    /etc/init.d/openvpn restart
+
+This will restart all configured openvpn tunnels.
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:67]:
```suggestion
This will restart all configured OpenVPN tunnels.
```

File: custom_openvpn_tunnel.rst [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+If you want to prevent the VPN from starting automatically when the firewall boots, you can disable it using the following commands: ::
+
+1. Disable the VPN in UCI: ::
+
+    uci set openvpn.myvpn.enabled='0'
+    uci commit openvpn
+
+2. Restart the running VPN tunnels (cause it's disabled now, it will stop, while other tunnels will do a full restart): ::
+
+    /etc/init.d/openvpn stop
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:95]: This is not correct: the proposed configuration completely disable the tunnel, not just at boot

File: custom_openvpn_tunnel.rst [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+
+Optional (2): Configure Authentication Credentials
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:98]:
```suggestion
Configure authentication credentials (optional)
```

File: custom_openvpn_tunnel.rst [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+
+1. Create a file named ``/etc/openvpn/myvpn.auth`` (replace ``myvpn`` with the VPN name if different): ::
+
+    vi /etc/openvpn/myvpn.auth
+
+2. Add the following content, replacing ``frank`` and ``frank_password`` with your username and password: ::
+                                    
+    user frank
+    password frank_password
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:111]:
```suggestion
    frank
    frank_password
```

File: custom_openvpn_tunnel.rst [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    chmod 600 /etc/openvpn/myvpn.auth
+    chown root:root /etc/openvpn/myvpn.auth
+                                    
+4. Update the OpenVPN configuration file (``myvpn.ovpn``) to reference the authentication file. Open the file for editing: ::
+   
+    vi /etc/openvpn/myvpn.ovpn
+
+5. Add or modify the following line: ::
+                                    
+    auth-user-pass /etc/openvpn/myvpn.auth
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:124]:
```suggestion
4. Update the OpenVPN configuration file (``myvpn.ovpn``) to reference the authentication file. ::
   
    echo "auth-user-pass /etc/openvpn/myvpn.auth" >>  /etc/openvpn/myvpn.ovpn

```

**Comment by filippocarletti** [comment]
For verification (step 5) I'd use the ps command and a suggestion to look into /var/log/messages for details.

**Comment by cotosso** [comment]
> For verification (step 5) I'd use the ps command and a suggestion to look into /var/log/messages for details.

@filippocarletti I added the section you indicated to the text. Pleae let me know.

File: custom_openvpn_tunnel.rst [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+You should see log entries indicating a successful connection.
+
+
+.. note:: 
+
+  - **File Name Consistency:** The configuration name ``myvpn`` must match the OpenVPN section name in UCI and the configuration file's location. If you change the name, ensure all references to ``myvpn`` in commands and filenames are updated.
+  - **Automatic Startup:** By setting ``enabled='1'``, the VPN client will automatically start whenever the firewall boots.
+
+
+Optional: Disable Automatic Startup
```
Comment (gsanchietti) [inline custom_openvpn_tunnel.rst:90]: I'd move this section to the end.
```suggestion
Disable the tunnel
```

Comment (cotosso) [reply custom_openvpn_tunnel.rst]: > I'd move this section to the end.

Do you mean after the section: "Configure authentication credentials (optional)"
or aty the very end after the firewall section?

Comment (gsanchietti) [reply custom_openvpn_tunnel.rst]: I'd say to the very end

**Comment by filippocarletti** [comment]
> > For verification (step 5) I'd use the ps command and a suggestion to look into /var/log/messages for details.
> 
> @filippocarletti I added the section you indicated to the text. Please let me know.

I'd use "ps -ef | grep myvpn"

**Comment by cotosso** [comment]
> I'd use "ps -ef | grep myvpn"

done

**Review by filippocarletti — CHANGES_REQUESTED** [review]

File: custom_openvpn_tunnel.rst (hunk 13 above)
Comment (cotosso) [reply custom_openvpn_tunnel.rst]: > I'd say to the very end

Section moved

File: custom_openvpn_tunnel.rst [hunk 14]
```diff
@@ -130,3 +130,47 @@ If the VPN requires a username and password, create an authentication file.
...
+To enable traffic through the VPN, it is necessary to configure the firewall on NethSecurity. 
+The best practice is to assign a fixed device name to the VPN, create a dedicated zone for the custom VPN, and associate the VPN device with that zone.
+
+
+1. Fix the VPN device name
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+To ensure the VPN device name remains consistent and avoids automatic assignment, it is crucial to fix the name in the OpenVPN configuration file. 
+Edit the file (``/etc/openvpn/myvpn.ovpn``) and include the following lines or modify them if they already exist (this example is made with a *routed* vpn): ::
+
+    -dev tun
```
[the remark this answers is not in the document]
Comment (cotosso) [reply custom_openvpn_tunnel.rst:134]: yes

File: custom_openvpn_tunnel.rst [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+-----------------------------------------------------
+
+To enable traffic through the VPN, it is necessary to configure the firewall on NethSecurity. 
+The best practice is to assign a fixed device name to the VPN, create a dedicated zone for the custom VPN, and associate the VPN device with that zone.
+
+
+1. Fix the VPN device name
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+To ensure the VPN device name remains consistent and avoids automatic assignment, it is crucial to fix the name in the OpenVPN configuration file. 
+Edit the file (``/etc/openvpn/myvpn.ovpn``) and include the following lines or modify them if they already exist (this example is made with a *routed* vpn): ::
```
Comment (filippocarletti) [inline custom_openvpn_tunnel.rst:132]:
```suggestion
Edit the file (``/etc/openvpn/myvpn.ovpn``) to change "dev tun" to "dev tunmyvpn" and add the following line (this example is made with a *routed* vpn): ::
```

File: custom_openvpn_tunnel.rst [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+
+    -dev tun
```
Comment (filippocarletti) [inline custom_openvpn_tunnel.rst:134]:
```suggestion

```

File: custom_openvpn_tunnel.rst [hunk 17]
```diff
@@ -0,0 +1,176 @@
...
+    +dev tunmyvpn
```
Comment (filippocarletti) [inline custom_openvpn_tunnel.rst:135]:
```suggestion

```

File: custom_openvpn_tunnel.rst [hunk 18]
```diff
@@ -0,0 +1,176 @@
...
+    +dev-type tun
```
Comment (filippocarletti) [inline custom_openvpn_tunnel.rst:136]:
```suggestion
    dev-type tun
```

**Review by filippocarletti — CHANGES_REQUESTED** [review]

File: custom_openvpn_tunnel.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (cotosso) [reply custom_openvpn_tunnel.rst]: Suggestions have been committed, a warning has beed added for interface's name maximum lenght.

**Review by filippocarletti — APPROVED** [review]

**Pull request merged**